cirq/google/optimizers/convert_to_sqrt_iswap.py

```diff
@@ -0,0 +1,290 @@
+# Copyright 2019 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import List, cast
+
+import numpy as np
+
+import cirq
+import sympy
+
+SQRT_ISWAP = cirq.ISWAP**-0.5
```

Comment:
Note that the device can handle both `cirq.ISWAP**0.5` and `cirq.ISWAP**-0.5` natively, since these are related by `ISWAP(a, b)**0.5 = Z(a) ISWAP(a, b)**-0.5 Z(b)`. This latter sequence of an iswap conjugated by Zs on one qubit appears in various places in the decompositions below, so we might be able to simplify things by defining:
```python
SQRT_ISWAP = cirq.ISWAP**0.5
SQRT_ISWAP_INV = cirq.ISWAP**-0.5
```
and then using both of these in the decompositions. This should avoid confusion about why we're choosing the negative exponent for "SQRT_ISWAP", and it also might make some of the decompositions more symmetrical.

Reply:
Done.